implementations/rust/ockam/ockam_api/src/nodes/service/manager.rs

```diff
@@ -142,13 +142,19 @@ impl NodeManager {
             _account_admin: None,
         };
 
+        let mut api_transport_flow_control_ids = vec![transport_options.tcp.flow_control_id];
+
+        if let Some(udp) = &transport_options.udp {
+            api_transport_flow_control_ids.push(udp.flow_control_id.clone());
+        }
+
         let mut s = Self {
             cli_state,
             node_name,
             node_identifier,
-            api_transport_flow_control_id: transport_options.api_transport_flow_control_id,
-            tcp_transport: transport_options.tcp_transport,
-            udp_transport: transport_options.udp_transport.clone(),
+            api_transport_flow_control_ids,
+            tcp_transport: transport_options.tcp.transport,
+            udp_transport: transport_options.udp.as_ref().map(|u| u.transport.clone()),
```

Comment:
I think you could avoid this clone with `transport_options.udp.take()` and get the reference from `s` below in line 177.